File: tests/expected/infer.cpp
#include <cassert>   // NOLINT(build/include_order)
#include <cstdint>   // NOLINT(build/include_order)
#include <iostream>  // NOLINT(build/include_order)
inline void foo() {
  int a = 10;
  int b = a;
  assert(b == 10);
  std::cout << b;
  std::cout << std::endl;
}

int main(int argc, char** argv) { foo(); }

File: tests/expected/infer.dart
// @dart=3.4
import 'package:sprintf/sprintf.dart';

foo() {
  final int a = 10;
  final int b = a;
  assert(b == 10);
  print(sprintf("%s", [b]));
}

main(List<String> argv) {
  foo();
}
